src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+ * seed construction, and
+ * {@link RandomGeneratorFactory#create()} is used for random seed
+ * construction. Example;
+ *
+ * <pre>{@code
+ *    RandomGeneratorFactory<RandomGenerator> factory = RandomGenerator.factoryOf("Random");
+ *
+ *     for (int i = 0; i < 10; i++) {
+ *         new Thread(() -> {
+ *             RandomGenerator random = factory.create(100L);
+ *             System.out.println(random.nextDouble());
+ *         }).start();
+ *     }
+ * }</pre>
+ *
+ * RandomGeneratorFactory also provides methods describing the attributes (or properties)
+ * of a generator and can be used to select random number generator algorithms.
+ * These methods are typically used in
+ * conjunction with {@link RandomGenerator#all()}. In this example, the code locates the
+ * {@link RandomGeneratorFactory} that produces {@link RandomGenerator RandomGenerators}
+ * with the highest number of state bits.
+ *
+ * <pre>{@code
+ *     RandomGeneratorFactory<RandomGenerator> best = RandomGenerator.all()
+ *         .sorted((f, g) -> Integer.compare(g.stateBits(), f.stateBits()))
+ *         .findFirst()
+ *         .orElse(RandomGenerator.factoryOf("Random"));
+ *     System.out.println(best.name() + " in " + best.group() + " was selected");
+ *
+ *     RandomGenerator rng = best.create();
+ *     System.out.println(rng.nextLong());
+ * }</pre>
+ *
+ * @since   16
+ *
+ * @see java.util.random
+ *
+ */
+public final class RandomGeneratorFactory<T extends RandomGenerator> {
+    /**
+     * Map of provider classes.
+     */
+    private static volatile Map<String, Provider<? extends RandomGenerator>> factoryMap;
+
+    /**
+     * Instance provider class of random number algorithm.
+     */
+    private final Provider<? extends RandomGenerator> provider;
+
+    /**
+     * Map of properties for provider.
+     */
+    private volatile Map<RandomGeneratorProperty, Object> properties = null;
+
+    /**
+     * Default provider constructor.
+     */
+    private volatile Constructor<T> ctor;
+
+    /**
+     * Provider constructor with long seed.
+     */
+    private Constructor<T> ctorLong;
+
+    /**
+     * Provider constructor with byte[] seed.
+     */
+    private Constructor<T> ctorBytes;
+
+    /**
+     * Private constructor.
+     *
+     * @param provider  Provider class to wrap.
+     */
+    private RandomGeneratorFactory(Provider<? extends RandomGenerator> provider) {
+        this.provider = provider;
+    }
+
+    /**
+     * Returns the factory map, lazily constructing map on first call.
+     *
+     * @return Map of RandomGeneratorFactory classes.
+     */
+    private static Map<String, Provider<? extends RandomGenerator>> getFactoryMap() {
+        Map<String, Provider<? extends RandomGenerator>> fm = RandomGeneratorFactory.factoryMap;
+        if (fm == null) {
+            synchronized (RandomGeneratorFactory.class) {
+                if (RandomGeneratorFactory.factoryMap == null) {
```

Comment:
if `RandomGeneratorFactory.factoryMap` is not null, it returns null because the value is not stored in fm.

Please use the class holder idiom (cf Effective Java)  instead of using the double-check locking pattern.

Reply:
? set in 148 and 152, but  class holder idiom +1

Reply:
If the second `RandomGeneratorFactory.factoryMap` return a non null value ...